src/draw/sw/lv_draw_sw_img.c

```diff
@@ -326,7 +326,12 @@ static void img_draw_core(lv_draw_unit_t * draw_unit, const lv_draw_image_dsc_t
         blend_dsc.src_buf = src_buf;
         blend_dsc.mask_buf = (lv_opa_t *)src_buf;
         blend_dsc.mask_buf += img_stride * src_w / header->w * src_h;
-        blend_dsc.mask_stride = src_w;
+        /**
+         * Note, for RGB565A8, lacking of stride parameter, we always use
+         * always half of RGB map stride as alpha map stride. The image should
+         * be generated in this way too.
+         */
+        blend_dsc.mask_stride = img_stride / 2;
```

Comment:
Should be updated [here](https://github.com/lvgl/lvgl/pull/4842/files#diff-242f305692436f289db7b3cdda6298157b9ab35835506b88b28a86b1524051abR391) too.

Reply:
Do you mean using `draw_buf_malloc` for `temp_buf`?

Reply:
I'm sorry, just oversight on my side. I thought `blend_dsc.mask_stride = blend_w;` is not correct, but it is.